src/module/system/damage/damage.ts

```diff
@@ -1,29 +1,21 @@
 import { StrikeSelf, AttackTarget } from "@actor/creature/types";
 import { DegreeOfSuccessString } from "@system/degree-of-success";
 import { BaseRollContext } from "@system/rolls";
-import { combineObjects } from "@util";
 
 /** The possible standard damage die sizes. */
-export const DAMAGE_DIE_FACES = new Set(["d4", "d6", "d8", "d10", "d12"] as const);
-export type DamageDieSize = SetElement<typeof DAMAGE_DIE_FACES>;
-
-export function nextDamageDieSize(dieSize: DamageDieSize) {
-    switch (dieSize) {
-        case "d4":
-            return "d6";
-        case "d6":
-            return "d8";
-        case "d8":
-            return "d10";
-        case "d10":
-            return "d12";
-        case "d12":
-            return "d12";
-    }
+const dieFacesTuple = ["d4", "d6", "d8", "d10", "d12"] as const;
+const DAMAGE_DIE_FACES = new Set(dieFacesTuple);
+type DamageDieSize = SetElement<typeof DAMAGE_DIE_FACES>;
+
+function nextDamageDieSize(next: { upgrade: DamageDieSize }): DamageDieSize;
+function nextDamageDieSize(next: { downgrade: DamageDieSize }): DamageDieSize;
+function nextDamageDieSize(next: { upgrade: DamageDieSize } | { downgrade: DamageDieSize }): DamageDieSize {
+    const [faces, direction] = "upgrade" in next ? [next.upgrade, 1] : [next.downgrade, -1];
+    return dieFacesTuple[dieFacesTuple.indexOf(faces) + direction] ?? faces;
```

Comment:
this is really more clever than it needs to be imo. `faces` should probably be called `dieSize`